jax/_src/api.py

```diff
@@ -1302,47 +1302,53 @@ def _split(x, indices, axis):
     return x.split(indices, axis)
 
 
-def vmap(fun: F, in_axes=0, out_axes=0, axis_name=None, axis_size=None) -> F:
+def vmap(fun: F,
+         in_axes: Union[None, int, Sequence[Any]] = 0,
+         out_axes: Any = 0,
+         axis_name: Optional[Hashable] = None,
+         axis_size: Optional[int] = None) -> F:
   """Vectorizing map. Creates a function which maps ``fun`` over argument axes.
 
   Args:
     fun: Function to be mapped over additional axes.
-    in_axes: An integer, None, or (nested) standard Python container
-      (tuple/list/dict) thereof specifying which input array axes to map over.
-
-      If each positional argument to ``fun`` is an array, then ``in_axes`` can
-      be an integer, a None, or a tuple of integers and Nones with length equal
-      to the number of positional arguments to ``fun``. An integer or ``None``
-      indicates which array axis to map over for all arguments (with ``None``
-      indicating not to map any axis), and a tuple indicates which axis to map
-      for each corresponding positional argument. Axis integers must be in the
-      range ``[-ndim, ndim)`` for each array, where ``ndim`` is the number of
-      dimensions (axes) of the corresponding input array.
-
-      If the positional arguments to ``fun`` are container (pytree) types, the
-      corresponding element of ``in_axes`` can itself be a matching container,
-      so that distinct array axes can be mapped for different container
-      elements. ``in_axes`` must be a container tree prefix of the positional
-      argument tuple passed to ``fun``. See this link for more detail:
-      https://jax.readthedocs.io/en/latest/pytrees.html#applying-optional-parameters-to-pytrees
+    in_axes: An optional integer--or a sequence of pytrees whose fields are
```

Comment:
So the benefit of the previous docstring is that it would be meaningful to users who are unfamiliar with JAX. Now it immediately jumps into a discussion that assumes an understanding of the term "pytree". I wonder if we could add the information about pytrees somewhere without making the entire discussion opaque to new users?

Reply:
Actually, I think the previous version does this fairly well: the bulk of the description uses familiar terms, and it only discusses pytrees in the last paragraph.